### PR TITLE
Upgrade to Lucene 6.2

### PR DESCRIPTION
**Pull request by zengmin2016**
need to make some changes to make it compile.
@javasoze

### CODE REVIEW

File: src/main/java/com/senseidb/clue/util/MatchSomeDocsQuery.java [hunk 1]
```diff
@@ -12,7 +12,17 @@
 public abstract class MatchSomeDocsQuery extends Query {
   
   protected abstract boolean match(int docId);
-  
+
```
Comment (javasoze) [inline src/main/java/com/senseidb/clue/util/MatchSomeDocsQuery.java]: why are these needed?

Comment (zengmin2016) [inline src/main/java/com/senseidb/clue/util/MatchSomeDocsQuery.java]: it won't compile without these, 'equals' and 'hashCode' are not abstract method in 6.2, so have to override them.
